caddyhttp/httpserver/context.go

```diff
@@ -410,6 +411,15 @@ func (c Context) RandomString(minLen, maxLen int) string {
 	return string(result)
 }
 
+// AddLink adds a link in response for future server push
+func (c Context) AddLink(link string) string {
+	if c.Res == nil {
+		return ""
+	}
+	c.Res.Header().Add("Link", "<"+link+">")
```

Comment:
Wouldn't you need rel=preload? See https://www.w3.org/TR/preload/#server-push-http-2